specs/monitoring_integration.adoc

```diff
@@ -0,0 +1,390 @@
+= Spec - Introducing a new component monitoring-integration
+
+A new component called Monitoring-Integration is introduced which integrates Grafana
+with tendrl core.
+
+
+== Problem description
+
+The new monitoring-integration component would primarily focus on dynamic
+configuration of grafana, based on the topology changes in etcd. Some of these
+responsibilities would be:
+
+* Monitoring-integration should store predefined json’s for dashboards, panels and alerts.
+* Configuring grafana authentication and authorization (https://github.com/Tendrl/specifications/pull/205),
+  based on changes in the primary tendrl AA scheme.
+* Keeping track of the topology changes in etcd to configure the grafana dashboards
+  to ensure that the correct nodes are part of the respective cluster dashboards.
+  (Not planned yet, for 1st milestone grafana is configured using some default
+  configuration from monitoring-integration)
+
+
+== Use Cases
+
+For 1st milestone:
+
+* Monitoring-integration should be able to configure grafana dashboard and panels.
+* Allow tendrl users to access grafana using anonymous authorization.
+
+For 2nd milestone:
+
+* Grafana alerting events should be notified via tendrl.
+* Monitoring-integration should be able to configure grafana based on topology
+  changes in etcd.
+* User should be able access grafana dashboard via tendrl user interface using
```

Comment:
This is a part of 3rd Milestone https://github.com/Tendrl/specifications/milestone/1